modules/apps/headless/headless-admin-user/headless-admin-user-test/src/testIntegration/java/com/liferay/headless/admin/user/resource/v1_0/test/UserGroupResourceTest.java

```diff
@@ -277,6 +291,79 @@ private UserGroup _postUserGroup(UserGroup userGroup) throws Exception {
 		return userGroupResource.postUserGroup(userGroup);
 	}
 
+	private void _testGetUserGroupWithNestedFields() throws Exception {
+		UserGroup postUserGroup = testGetUserGroup_addUserGroup();
+
+		User user1 = UserTestUtil.addUser();
+		User user2 = UserTestUtil.addUser();
+		User user3 = UserTestUtil.addUser();
+
+		Long[] userIds = {user1.getUserId(), user2.getUserId()};
+
+		userGroupResource.postUserGroupUsers(postUserGroup.getId(), userIds);
+
+		UserGroupResource userGroupResource = UserGroupResource.builder(
+		).authentication(
+			"test@liferay.com", PropsValues.DEFAULT_ADMIN_PASSWORD
+		).locale(
+			LocaleUtil.getDefault()
+		).parameters(
+			"nestedFields", "creator,userAccountBriefs"
+		).build();
+
+		UserGroup getUserGroup = userGroupResource.getUserGroup(
+			postUserGroup.getId());
+
+		Creator creator = getUserGroup.getCreator();
+
+		Assert.assertTrue(creator.getId() == TestPropsValues.getUserId());
+
+		UserAccountBrief[] userAccountBriefs =
+			getUserGroup.getUserAccountBriefs();
+
+		Assert.assertTrue(
+			ArrayUtil.contains(
+				userAccountBriefs,
+				new UserAccountBrief() {
+					{
+						setAlternateName(user1.getScreenName());
+						setEmailAddress(user1.getEmailAddress());
+						setExternalReferenceCode(
+							user1.getExternalReferenceCode());
+						setId(user1.getUserId());
+						setName(user1.getFullName());
+					}
+				}));
+
+		Assert.assertTrue(
+			ArrayUtil.contains(
+				userAccountBriefs,
+				new UserAccountBrief() {
+					{
+						setAlternateName(user2.getScreenName());
+						setEmailAddress(user2.getEmailAddress());
+						setExternalReferenceCode(
+							user2.getExternalReferenceCode());
+						setId(user2.getUserId());
+						setName(user2.getFullName());
+					}
+				}));
+
+		Assert.assertFalse(
+			ArrayUtil.contains(
+				userAccountBriefs,
+				new UserAccountBrief() {
+					{
+						setAlternateName(user3.getScreenName());
+						setEmailAddress(user3.getEmailAddress());
+						setExternalReferenceCode(
+							user3.getExternalReferenceCode());
+						setId(user3.getUserId());
+						setName(user3.getFullName());
+					}
+				}));
+	}
```

Comment:
can we just check for the ids? It should be possible to have a function as a parameter and verify if the brief.getId = user1 or user2 and not user3